src/wp-includes/theme.php

```diff
@@ -873,6 +873,10 @@ function switch_theme( $stylesheet ) {
 	$wp_stylesheet_path = null;
 	$wp_template_path   = null;
 
+	// Clear pattern caches only on single as a this cache maybe being used on other sites.
+	if ( ! is_multisite() ) {
+		delete_site_option( 'wp_theme_patterns_' . $old_theme->get_stylesheet() );
+	}
```

Comment:
See above - additionally though, we also should clear the old parent theme's cache if any. This is cheap enough to leave it unconditional, similar to how it is used further above in this function to clear paused themes.
```suggestion
	// Clear pattern caches to not waste database space.
	delete_transient( 'wp_theme_patterns_' . $old_theme->get_stylesheet() );
	delete_transient( 'wp_theme_patterns_' . $old_theme->get_template() );
```